cmake/CompilerOptions.cmake

```diff
@@ -17,6 +17,23 @@ if(${ARBDEV_COLOR})
     add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:${colorflags}>")
 endif()
 
+# A library to collect compiler-specific linking adjustments.
+
+add_library(arbor-compiler-compat INTERFACE)
+# TODO Remove when upgrading GCC.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
+    target_link_libraries(arbor-compiler-compat INTERFACE stdc++fs)
+  endif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
+endif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+# TODO Remove when upgrading Clang
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+    target_link_libraries(arbor-compiler-compat INTERFACE stdc++fs)
+  endif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+endif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+install(TARGETS arbor-compiler-compat EXPORT arbor-targets)
+
```

Comment:
Bit of a style suggestion: Can we remove the condition argument from `endif`? I believe that's more consistent with the rest of our CMake code.

Reply:
fixed